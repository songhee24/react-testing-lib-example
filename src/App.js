import React, { useEffect } from "react";

function App() {
  const [data, setData] = React.useState(null);

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 500);
  }, []);

  return (
    <div>
      {data && <div style={{ color: "red" }}>data</div>}
      <h1>Hello World</h1>
      <button>click me</button>
      <input type="text" placeholder="input value..." />
    </div>
  );
}

export default App;
