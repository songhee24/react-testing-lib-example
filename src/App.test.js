import { render, screen } from "@testing-library/react";
import App from "./App";

describe("React Core Test", () => {
  test("renders hello world element", () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const buttonElem = screen.getByRole("button");
    const inputElem = screen.getByPlaceholderText(/input value/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(buttonElem).toBeInTheDocument();
    expect(inputElem).toMatchSnapshot();
    // screen.debug()
  });

  test("screen methods", async () => {
    render(<App />);
    // const helloWorldElem = screen.queryByText(/hello world2/i)
    // expect(helloWorldElem).toBeNull()
    screen.debug();
    const helloWorldElem = await screen.findByText(/data/i);
    expect(helloWorldElem).toBeInTheDocument();
    screen.debug();
  });
});
