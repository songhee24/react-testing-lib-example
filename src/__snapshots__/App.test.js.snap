// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hello world element 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
